cli/src/cmd/forge/verify.rs

```diff
@@ -23,18 +23,54 @@ use foundry_utils::Retry;
 use futures::FutureExt;
 use once_cell::sync::Lazy;
 use regex::Regex;
+use reqwest::get;
 use semver::{BuildMetadata, Version};
 use std::{
     collections::BTreeMap,
+    fmt::{Display, Formatter},
     path::{Path, PathBuf},
+    str::FromStr,
 };
+
 use tracing::{trace, warn};
 
 pub static RE_BUILD_COMMIT: Lazy<Regex> =
     Lazy::new(|| Regex::new(r#"(?P<commit>commit\.[0-9,a-f]{8})"#).unwrap());
 
 pub const RETRY_CHECK_ON_VERIFY: RetryArgs = RetryArgs { retries: 6, delay: Some(10) };
 
+#[derive(Debug, Clone)]
+pub enum VerificationProvider {
+    Etherscan,
+    Sourcify,
+}
+
+impl FromStr for VerificationProvider {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "e" | "etherscan" => Ok(VerificationProvider::Etherscan),
+            "s" | "sourcify" => Ok(VerificationProvider::Sourcify),
+            _ => Err(format!("Unknown field: {s}")),
+        }
+    }
+}
+
+impl Display for VerificationProvider {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            VerificationProvider::Etherscan => {
+                write!(f, "Etherscan")?;
+            }
+            VerificationProvider::Sourcify => {
+                write!(f, "Sourcify")?;
+            }
+        };
+        Ok(())
+    }
+}
+
```

Comment:
let's move these below the `VerifyArgs`, so each cmd file starts with the actual command